google-cloud-spanner/src/test/java/com/google/cloud/spanner/spi/v1/SpannerRpcMetricsTest.java

```diff
@@ -0,0 +1,303 @@
+import java.net.InetSocketAddress;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.junit.After;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class SpannerRpcMetricsTest {
+
+  private static MockSpannerServiceImpl mockSpanner;
+  private static Server server;
+  private static InetSocketAddress address;
+  private static Spanner spannerWithOpenTelemetry;
+  private static DatabaseClient databaseClient;
+  private static final Map<SpannerRpc.Option, Object> optionsMap = new HashMap<>();
+  private static MockSpannerServiceImpl mockSpannerNoHeader;
+  private static Server serverNoHeader;
+  private static InetSocketAddress addressNoHeader;
+  private static Spanner spannerNoHeaderNoOpenTelemetry;
+  private static DatabaseClient databaseClientNoHeader;
+  private static String instanceId = "fake-instance";
+  private static String databaseId = "fake-database";
+  private static String projectId = "fake-project";
+  private static AtomicInteger fakeServerTiming = new AtomicInteger(new Random().nextInt(1000) + 1);
+  private static final Statement SELECT1AND2 =
+      Statement.of("SELECT 1 AS COL1 UNION ALL SELECT 2 AS COL1");
+  private static final ResultSetMetadata SELECT1AND2_METADATA =
+      ResultSetMetadata.newBuilder()
+          .setRowType(
+              StructType.newBuilder()
+                  .addFields(
+                      StructType.Field.newBuilder()
+                          .setName("COL1")
+                          .setType(
+                              com.google.spanner.v1.Type.newBuilder()
+                                  .setCode(TypeCode.INT64)
+                                  .build())
+                          .build())
+                  .build())
+          .build();
+  private static final com.google.spanner.v1.ResultSet SELECT1_RESULTSET =
+      com.google.spanner.v1.ResultSet.newBuilder()
+          .addRows(
+              ListValue.newBuilder()
+                  .addValues(com.google.protobuf.Value.newBuilder().setStringValue("1").build())
+                  .build())
+          .addRows(
+              ListValue.newBuilder()
+                  .addValues(com.google.protobuf.Value.newBuilder().setStringValue("2").build())
+                  .build())
+          .setMetadata(SELECT1AND2_METADATA)
+          .build();
+  private static final Statement UPDATE_FOO_STATEMENT =
+      Statement.of("UPDATE FOO SET BAR=1 WHERE BAZ=2");
+
+  private static InMemoryMetricReader inMemoryMetricReader;
+
+  private static InMemoryMetricReader inMemoryMetricReaderInjected;
+
+  @BeforeClass
+  public static void startServer() throws IOException {
+    assumeFalse(EmulatorSpannerHelper.isUsingEmulator());
```

Comment:
Why is this test not supported when the emulator is used? (This is not an integration test, so it would normally not interact with the emulator)